pandas/tests/computation/test_eval.py

```diff
@@ -38,6 +39,23 @@
 _scalar_skip = 'in', 'not in'
 
 
+@pytest.fixture(params=[
+    pytest.mark.skipif(engine == 'numexpr' and not _USE_NUMEXPR,
+                       reason='numexpr enabled->{enabled}, '
+                              'installed->{installed}'.format(
+                            enabled=_USE_NUMEXPR,
+                            installed=_NUMEXPR_INSTALLED))(engine)
```

Comment:
@MaximilianR There is a slight difference in your method; the one I have will mark the skip at collection time, while yours requires running the fixture to find out about the skip. It's a rather minor difference though.

Also, @jreback your first code block works fine. In fact, `pytest` will understand the generator, so you don't need the `list`:
```python
def create_things():
    return range(5)

@pytest.mark.parametrize('t', create_things())
def test_thing(t):
    ...
```
works fine.